src/Jobs/TileSingleOverlay.php

```diff
@@ -0,0 +1,145 @@
+use RecursiveIteratorIterator;
+use VipsImage;
+
+class TileSingleOverlay extends Job implements ShouldQueue
+{
+    use InteractsWithQueue, SerializesModels;
+
+    /**
+     * The overlay to generate tiles for.
+     *
+     * @var GeoOverlay
+     */
+    public $overlay;
+
+    /**
+     * Path to the temporary storage file for the tiles.
+     *
+     * @var string
+     */
+    public $tempPath;
+
+    public $file;
+
+    /**
+     * Ignore this job if the overlay does not exist any more.
+     *
+     * @var bool
+     */
+    protected $deleteWhenMissingModels = true;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param GeoOverlay $overlay The Overlay to generate tiles for.
+     *
+     * @return void
+     */
+    public function __construct(GeoOverlay $overlay)
+    {
+        $this->overlay = $overlay;
+        $this->tempPath = config('overlay.tiles.tmp_dir')."/{$overlay->id}";
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        try {
+            $disk = config('geo.overlay_storage_disk');
+            $this->file = new GenericFile("{$disk}://{$this->overlay->getPathAttribute()}");
+            FileCache::getOnce($this->file, [$this, 'generateTiles']);
+            $this->uploadToStorage();
+            $this->overlay->tilingInProgress = false;
+            $this->overlay->save();
+        } finally {
+            File::deleteDirectory($this->tempPath);
+        }
+    }
+
+    /**
+     * Generate tiles for the image and put them to temporary storage.
+     *
+     * @param string $path Path to the cached image file.
+     */
+    public function generateTiles(GenericFile $file, $path)
+    {
+        $this->getVipsImage($path)->dzsave($this->tempPath, [
+            'layout' => 'zoomify',
+            'container' => 'fs',
+            'strip' => true,
+        ]);
+    }
+
+    /**
+     * Upload the tiles from temporary local storage to the tiles storage disk.
+     */
+    public function uploadToStorage()
+    {
+        // +1 for the connecting slash.
+        $prefixLength = strlen($this->tempPath) + 1;
+        $iterator = $this->getIterator($this->tempPath);
+        $disk = Storage::disk(config('geo.overlay_storage_disk'));
+        $fragment = $this->overlay->id;
+        try {
+            foreach ($iterator as $pathname => $fileInfo) {
+                echo "pathname: " . $pathname . "<br>";
+                echo "fileInfo: " . $fileInfo . "<br>";
+                echo "disk path: " . $this->overlay->volume_id . "/" . $fragment . "<br>";
+                $disk->putFileAs("{$this->overlay->volume_id}/{$fragment}", $fileInfo, substr($pathname, $prefixLength));
+            }
+        } catch (Exception $e) {
+            $disk->deleteDirectory($fragment);
+            throw $e;
+        }
+    }
+
+    /**
+     * Get the vips image instance.
+     *
+     * @param string $path
+     *
+     * @return \Jcupitt\Vips\Image
+     */
```

Comment:
Hi @mzur, I keep getting the error 

> Call to undefined function Jcupitt\Vips\vips_image_new_from_file()

Initially I thought this has to do with the missing php-package ext-vips, so I updated PHP to a newer version (8.3), included the vips-package, and reinstalled all dependencies from composer.json. But still, the issue remains.  
I also tried things like changing the import from `use VipsImage` to `use Jcupitt\Vips` and then `Vips/Image::newFromFile()` with no success either. I clearly see that the jcupitt package is in the biigle/vendor directory, so I don't know what else to check at the moment...

Reply:
Do you get the error while running the app/tests in the Docker containers?

Reply:
I ran all tests (`composer test`) and did not get the error.

Reply:
So where/when *do* you get the error?

Reply:
I get the error after trying to upload a geoTIFF, i.e. when the TileSingleOverlay Job is called within the VolumeGeoOverlayController.

Reply:
Do you execute the job synchronously in the controller? Vips is only installed in the worker container (processing queued jobs), not the app container (processing the HTTP requests).